pkg/kubelet/prober/prober_test.go

```diff
@@ -333,3 +337,127 @@ func TestNewExecInContainer(t *testing.T) {
 		}
 	}
 }
+
+func TestNewProber(t *testing.T) {
+	runner := &containertest.FakeContainerCommandRunner{}
+	recorder := &record.FakeRecorder{}
+	prober := newProber(runner, recorder)
+
+	assert.NotNil(t, prober, "Expected prober to be non-nil")
+	assert.Equal(t, runner, prober.runner, "Expected prober runner to match")
+	assert.Equal(t, recorder, prober.recorder, "Expected prober recorder to match")
+
+	assert.NotNil(t, prober.exec, "exec probe initialized")
+
+	assert.NotNil(t, prober.http, "http probe initialized")
+
+	assert.NotNil(t, prober.tcp, "tcp probe initialized")
+
+	assert.NotNil(t, prober.grpc, "grpc probe initialized")
+
```

Comment:
nit: Since we're checking similar things, I think it's fine to group them together without line breaks.

```suggestion
	assert.NotNil(t, prober.exec, "exec probe initialized")
	assert.NotNil(t, prober.http, "http probe initialized")
	assert.NotNil(t, prober.tcp, "tcp probe initialized")
	assert.NotNil(t, prober.grpc, "grpc probe initialized")
```